### ISSUE TITLE
🐛 user-provided comparison function does not correctly implement a total order

### ISSUE THREAD

**Issue by oliversalzburg** [biomejs/biome#5756]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         yarn/4.9.1
```

### What happened?

```shell
yarn biome check --write --no-errors-on-unmatched
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: library/core/src/slice/sort/shared/smallsort.rs:865:5
Thread Name: biome::workspace_worker_10
Message: user-provided comparison function does not correctly implement a total order

/home/oliver/projects/kitten-science/themes/contrib/pal.json internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: user-provided comparison function does not correctly implement a total order
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
```

The referenced `pal.json` is: <https://github.com/kitten-science/themes/blob/0f69d9ed05accc42270fcdb6ffce1298b7fe92c7/contrib/pal.json>

Relates to #5700 

### Expected result

Problem did not exist in pre-2.0.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by oliversalzburg** [comment]
Seems to relate to having `useSortedKeys` enabled in the config, but I didn't have much time to look into it. I copied the config from <https://github.com/kitten-science/kitten-scientists/blob/main/biome.json>.

**Comment by AndreiTS** [comment]
@ematipico I'm facing the same issue in version 2.0.5:

```
Biome encountered an unexpected error
This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.
When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:
Source Location: library/core/src/slice/sort/shared/smallsort.rs:860:5
Thread Name: biome::workspace_worker_0
Message: user-provided comparison function does not correctly implement a total order
```
```
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v24.2.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.12.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0

```

I don't know what code is causing this, it would be helpful if biome told me

**Comment by AndreiTS** [comment]
The output looks like this:
![Image](https://github.com/user-attachments/assets/ee0d3f1b-e011-4c53-8021-a827c25b1702)

**Comment by Conaclos** [comment]
@AndreiTS Could you open a new issue with a code and config that reproduces the issue?
